Assets/Scripts/Interface/AlertMeter.cs

```diff
@@ -0,0 +1,187 @@
+            if (currentState == AlertState.Normal && fillAmount >= 1)
+            {
+                ChangeAlertState(AlertState.Activated);
+            }
+            if (currentState == AlertState.Activated && fillAmount <= 0)
+            {
+                ChangeAlertState(AlertState.Normal);
+            }
+        }
+
+        private void ChangeAlertState(AlertState alertState)
+        {
+            currentState = alertState;
+            if (alertState == AlertState.Normal)
+            {
+                fillAmount = 0;
+            }
+            else
+            {
+                StartCoroutine(ShakeRadio());
+                StartCoroutine(InflateRadio());
+            }
+        }
+
+        private IEnumerator ShakeRadio()
+        {
+            while (currentState == AlertState.Activated)
+            {
+                radioShineRect.localRotation *= Quaternion.Euler(Vector3.forward * 600 * Time.deltaTime);
+                alertMeterRect.localRotation = Quaternion.Euler(new Vector3(alertMeterRect.localRotation.x, alertMeterRect.localRotation.y, Random.Range(-4, 4)));
+                yield return null;
+            }
+            radioShineRect.localRotation = Quaternion.Euler(new Vector3(radioShineRect.localRotation.x, radioShineRect.localRotation.y, 0));
+            alertMeterRect.localRotation = Quaternion.Euler(new Vector3(radioShineRect.localRotation.x, radioShineRect.localRotation.y, 0));
+            radioShine.SetActive(false);
+        }
+
+        private IEnumerator InflateRadio()
+        {
+            // Define o tempo que ser� tomado em cada etapa da anima��o
+            float timeStart = 0.1f;
+            float timeHold = 0f;
+            float timeEnd = 0.2f;
+            Vector3 initScale = alertMeterRect.localScale;
+
+            for (float i = 0; i < timeStart; i += 0)
+            {
+                float scale = i / timeStart;
+                scale = scale * scale * scale;
+                alertMeterRect.localScale = initScale * Mathf.Lerp(1, 1.3f, scale);
+                i += Time.deltaTime;
+                yield return null;
+            }
+            radioShine.SetActive(true);
+            yield return new WaitForSeconds(timeHold);
+            for (float i = timeEnd; i > 0; i += 0)
+            {
+                float scale = i / timeEnd;
+                scale = scale * scale * scale;
+                alertMeterRect.localScale = initScale * Mathf.Lerp(1, 1.3f, scale);
+                i -= Time.deltaTime;
+                yield return null;
+            }
+        }
+
+        private GameObject GetChildGameObject(GameObject fromGameObject, string withName)
+        {
+            // Author: Isaac Dart, June-13.
+            Transform[] ts = fromGameObject.transform.GetComponentsInChildren<Transform>(true);
+            foreach (Transform t in ts) if (t.gameObject.name == withName) return t.gameObject;
+            return null;
+        }
```

Comment:
No geral, ficou bem legal esse script, so faltou um metodo publico que permita alguem controlar essa UI:
```suggestion
        }
        public void SetAlertLevel(float _alertLevel)
        {
            if (_alertLevel > 1.0f || _alertLevel < 0)
            {
                Debug.LogError("Invalid alert level!!!");
                return;
            }
            fillAmount = _alertLevel;
        }
```

Reply:
blz, agradeço já mandar kk